### ISSUE TITLE
SelectWithIndex (or something similar)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [blyzer/morelinq#29]
```
I've seen a number of questions like this one: 
http://stackoverflow.com/questions/1593235/linq-to-xml-dictionary-conversion

Where the overload for Select taking a `Func<TSource,int,TResult>` is 
basically used to create key/value pairs. We should make this easier with a 
single method call - either using KeyValuePair or a new IndexValuePair 
struct.
```

Original issue reported on code.google.com by `jonathan.skeet` on 20 Oct 2009 at 8:52

**Comment by GoogleCodeExporter** [comment]
```
SelectWithIndex is, IMO, the existing overload of Select.
What he was asking sounds to me to be an overload of ToDictionary where the Key 
selector includes an index parameter.
```

Original comment by `tormod.s...@gmail.com` on 16 Jun 2012 at 11:37

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 07ebb1349a1d
The operator is simply called Index.
```

Original comment by `azizatif` on 16 Jun 2012 at 4:28
- Changed state: **Fixed**
- Added labels: **Milestone-Release1.0**

**Comment by GoogleCodeExporter** [comment]
Original comment by `azizatif` on 16 Jun 2012 at 4:29
